utils.d.ts

```diff
@@ -0,0 +1,23 @@
+declare global {
```

Comment:
These types will be accessible everywhere without needing to import them. For example, you can use `DeepPartial` the same way you can use the `Partial` utility type from TS.